### PR TITLE
Feat/biome

### PR DESCRIPTION
**Pull request by soofstad**
Remove old Yarn2(PnP) stuff

Replaces `eslint` and `prettier` with `biome`

Remove black, as ruff does the same and more. 
Change `ruff` pre-commit hook to official repo.

### CODE REVIEW

**Comment by sebastianvitterso** [comment]
image

My guess is that you need to use a newer node image for the runner, or use the `actions/setup-node@v4` action.

File: web/src/App.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ function App() {
     return (
       <CenterContainer>
         <Typography>Login in progress.</Typography>
-        <Progress.Dots color="primary" />
+        <Progress.Dots color='primary' />
```
Comment (sebastianvitterso) [inline web/src/App.tsx:35]: Vote for having double fnutts here, so changing `"jsxQuoteStyle": "single",` to `"jsxQuoteStyle": "double",` in `biome.json`.

File: web/biome.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+  "$schema": "https://biomejs.dev/schemas/1.8.0/schema.json",
+  "files": {
+    "ignore": ["documentation/*", "api/*", "web/src/api/generated"]
+  },
+  "javascript": {
+    "formatter": {
+      "indentStyle": "space",
+      "indentWidth": 2,
+      "jsxQuoteStyle": "single",
```
Comment (sebastianvitterso) [inline web/biome.json:10]:
```suggestion
      "jsxQuoteStyle": "double",
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -37,58 +37,22 @@ repos:
       - id: conventional-pre-commit
         stages: [commit-msg]
 
-  - repo: https://github.com/charliermarsh/ruff-pre-commit
-    rev: "v0.0.289"
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: "v0.4.7"
     hooks:
```
Comment (jorgenengelsen) [inline .pre-commit-config.yaml:42]: To replace black as a formatter I think you need both:
```
- id: ruff-format
        name: Code formatting
        files: ^src/.*\.py$

  - id: ruff
      name: Code linting
      files: ^src/.*\.py$
      args:
        - --fix

File: web/Dockerfile [hunk 4]
```diff
@@ -32,7 +32,7 @@ RUN touch /var/run/nginx.pid && \
 USER 1000
 
 # Fails to build styled-common with node 18
```
Comment (sebastianvitterso) [inline web/Dockerfile:34]: Guess this comment is outdated to, but otherwise I approve!

**Review by sebastianvitterso — APPROVED** [review]
Nice!

**Pull request merged**